server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyToResponseBody.scala

```diff
@@ -30,7 +30,16 @@ class NettyToResponseBody extends ToResponseBody[ByteBuf, NoStreams] {
         val stream = v.asInstanceOf[InputStream]
         Unpooled.wrappedBuffer(stream.readAllBytes())
 
-      case RawBodyType.FileBody         => Unpooled.wrappedBuffer(Files.readAllBytes(v.file.toPath))
+      case RawBodyType.FileBody         =>
+        val fileRange = v.asInstanceOf[FileRange]
+        val bytes = (for {
+          range <- fileRange.range
+          start <- range.start
+          end <- range.end
+        } yield Files.readAllBytes(fileRange.file.toPath).slice(start.toInt, end.toInt + NettyToResponseBody.IncludingLastOffset))
+          .getOrElse(Files.readAllBytes(fileRange.file.toPath))
```

Comment:
can we implement this without reading the whole content into memory?

Reply:
sure we can... I've expected it to be more straightforward though